include/boost/asio/windows_sspi/detail/sspi_functions.hpp

```diff
@@ -0,0 +1,127 @@
+//
+// windows_sspi/detail/sspi_functions.hpp
+// ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+//
+// Copyright (c) 2020 Kasper Laudrup (laudrup at stacktrace dot dk)
+//
+// Distributed under the Boost Software License, Version 1.0. (See accompanying
+// file LICENSE_1_0.txt or copy at http://www.boost.org/LICENSE_1_0.txt)
+//
+
+#ifndef BOOST_ASIO_WINDOWS_SSPI_DETAIL_SSPI_FUNCTIONS_HPP
+#define BOOST_ASIO_WINDOWS_SSPI_DETAIL_SSPI_FUNCTIONS_HPP
+
+// TODO: Avoid cluttering global namespace (and avoid Windows headers if possible)
+#define SECURITY_WIN32
+#include <security.h>
+
+namespace {
```

Comment:
Unnamed namespaces in headers are a recipe for ODR violations, because functions defined with non-internal linkage will have a different definition of the entities in that namespace in each TU that includes this header.

Reply:
Of course. Thanks for pointing this out. Guess it would be OK to move this out of the anonymous namespace? It is still in a `detail` namespace after all?

Reply:
You can assume that users will never use identifiers in any `detail` namespace (no matter where it is nested in your headers.